packages/dds/tree/src/feature-libraries/defaultFieldKinds.ts

```diff
@@ -454,9 +489,15 @@ export interface OptionalChangeset {
 	fieldChange?: OptionalFieldChange;
 
 	/**
-	 * Changes to the node which will be in the field after this changeset is applied.
+	 * Changes to the node which were in the field before this changeset is applied, or the node deleted in this field in the given revision
 	 */
 	childChange?: NodeChangeset;
+
+	/**
+	 * The revision the node `childChange` is referring to was deleted in.
+	 * If undefined, `childChange` refers to the node currently in this field.
+	 */
+	deletedBy?: RevisionTag;
```

Comment:
For the sake of representing a single change, these degrees of freedom do seem sufficient.

That doesn't seem to be the case for representing the potental output of a transaction though. For example, a transaction could edit node foo, replace node foo with node bar, then edit node bar. Is being able to represent such things a non-goal for this PR?

Reply:
You're correct about what edits we can represent. Handling edits to nodes which only transiently existed during a transaction is not a goal of this PR.

Reply:
Cool. Please add some comment to explain what is and isn't representable.

Reply:
I've added a comment with the following text under the comments for `deletedBy`: 

"This representation is sufficient for representing changes to the node present before this changeset and
after this changeset, but not for changes to nodes that existed only transiently in a transaction."